grails-app/domain/org/pih/warehouse/shipping/Shipment.groovy

```diff
@@ -697,5 +697,63 @@ class Shipment implements Comparable, Serializable {
     Boolean hasChildContainer() {
         shipmentItems?.any { it?.container }
     }
+
+    Set<Event> getCustomEvents() {
+        return events.findAll { it.eventType?.eventCode?.name()?.startsWith("CUSTOMS") }
+    }
+
+    @Override
+    ReferenceDocument getReferenceDocument() {
+        return new ReferenceDocument(
+                label: shipmentNumber,
+                url: "/stockMovement/show/${requisition?.id ?: id}",
+                id: id,
+                identifier: shipmentNumber,
+                description: description,
+                name: name,
+        )
+    }
+
+    @Override
+    List<HistoryItem> getHistory() {
+        List<HistoryItem> histories = []
+        ReferenceDocument shipmentReferenceDocument = getReferenceDocument()
+        // First collect history of CREATED event
+        histories.add(new HistoryItem<Shipment>(
+                identifier: shipmentNumber,
+                date: dateCreated,
+                location: origin,
+                eventCode: EventCode.CREATED,
+                referenceDocument: shipmentReferenceDocument
```

Comment:
This is going to be weird looking, but if you remove Line 720 and change this line to 
```
referenceDocument: referenceDocument
```
it should work.

Reply:
yeah it will, I think I kept that, because initially I had a few reference documents in the `getHistory()` method, before I decided to move each creator to a proper domain (Shipment, Receipt).